NitroxClient/GameLogic/ItemContainers.cs

```diff
@@ -13,86 +13,100 @@
 using NitroxModel_Subnautica.DataStructures;
 using UnityEngine;
 
-namespace NitroxClient.GameLogic
+namespace NitroxClient.GameLogic;
+
+public class ItemContainers
 {
-    public class ItemContainers
+    private readonly IPacketSender packetSender;
+    private readonly EntityMetadataManager entityMetadataManager;
+    private readonly Items items;
+
+    public ItemContainers(IPacketSender packetSender, EntityMetadataManager entityMetadataManager, Items items)
     {
-        private readonly IPacketSender packetSender;
-        private readonly EntityMetadataManager entityMetadataManager;
+        this.packetSender = packetSender;
+        this.entityMetadataManager = entityMetadataManager;
+        this.items = items;
+    }
 
-        public ItemContainers(IPacketSender packetSender, EntityMetadataManager entityMetadataManager)
+    public void BroadcastItemAdd(Pickupable pickupable, Transform containerTransform, ItemsContainer container)
+    {
+        // We don't want to broadcast that event if it's from another player's inventory
+        if (containerTransform.GetComponentInParent<RemotePlayerIdentifier>(true))
         {
-            this.packetSender = packetSender;
-            this.entityMetadataManager = entityMetadataManager;
+            return;
         }
 
-        public void BroadcastItemAdd(Pickupable pickupable, Transform containerTransform)
+        if (!InventoryContainerHelper.TryGetOwnerId(containerTransform, out NitroxId ownerId))
         {
-            // We don't want to broadcast that event if it's from another player's inventory
-            if (containerTransform.GetComponentInParent<RemotePlayerIdentifier>(true))
-            {
-                return;
-            }
+            // Error logging is done in the function
+            return;
+        }
 
-            if (!pickupable.TryGetIdOrWarn(out NitroxId itemId))
-            {
-                return;
-            }
+        // For planters, we'll always forcefully recreate the entity to ensure there's no desync
+        if (container.containerType == ItemsContainerType.LandPlants || container.containerType == ItemsContainerType.WaterPlants)
```

Comment:
Is it worth us having Plants as a type then inherit? Then we can only have one check here and work for all plants

Reply:
It is not a valid general case because the items that need to go through this don't necessarily inherit from a "plant" related behaviour. Thus the easiest check is on the container type